### DISCUSSION TITLE
UI : Expose a toggle at the bottom of the system tray icon menu

### DISCUSSION THREAD

**Discussion by cassepipe** [blueman-project/blueman#2049] [Ideas]
I suppose there must a exit a toggle UI object like those somewhere in some python UI library :

![image](https://user-images.githubusercontent.com/42150430/227494118-5c565a36-e157-4b6b-a1a0-59d0b55a4b15.png)

It'd be great if the icon tray would expose a toggle on hover/click or, if that's complicated, have this toggle element be at the base of the icon's context menu so bluetooth can be turned on and off in two quick clicks.

Current UI:
   Acivation : 
      - 2 left clicks, half a screen apart from each other
      - 1 right click and 1 left click apart from each other
   Deactivation:
     - 1 right click and 1 left click apart from each other
     - No discoverability, there's no option on the main window
  
When I have to turn off bluetooth I generally choose `Exit`, because it's closer :)

Curious to hear your opinions

**Comment by cschramm** [comment]
I guess it makes sense to have the enable / disable entry as close as possible to the tray icon, yes. However, we cannot easily anticipate which item is the closest to the tray icon. I assume your tray is at the bottom of desktop so that the menu opens above. If it would be at the top and the menu would open below, the first icon would actually be the closest.

I'm wondering about your actual use case, though.

> When I have to turn off bluetooth I generally choose Exit, because it's closer :)

`Exit` just terminates blueman. It does not disable Bluetooth in any way.

**Reply by cassepipe** [reply]
1. Uh didn't think of that oops. Yes, bottom panel indeed. On Cinnamon. Since the mint team has offcially adopted `blueman` as its bluetooth manager maybe they could make the panel API expose its position in order to adapt the menu ?

2. Good to know thanks, I'll make sure to turn the bluetooth off for real

3.  Whatever its orientation of the panel, a change of the bluetooth icon into a toggle would make everyone happy I believe. Is that a hard thing to do ?

**Reply by cschramm** [reply]
I'm not sure where you want to put that toggle. If you want to replace the power switching menu item, it's just a matter of presentation and I would not expect a menu item (that closes the menu) to look like a toggle. I also don't see a reason to keep the menu open, especially when bluetooth gets disabled. If you want to replace the tray icon itself, it would also have to act like a toggle and it does not. It opens the manger on left-click (if possible) and its menu on right-click. Nothing toggle-y there.
